# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fetch/InternalResponse.cpp
# Commit: dbdca4661a35
# Full Hash: dbdca4661a35c55d343cb76fd10366cdbd1e4c09
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-01-18 09:51:31
# Regressor Bug: 1351231
# File Overlap Count: 2
# Description:
#   Bug 1351231 - PFetch protocol declaration and implementation. r=dom-worker-reviewers,jesup
#   
#   Depends on D138813
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D142436
# ==============================================================================

diff -r ea27cd66fefd -r dbdca4661a35 dom/fetch/InternalResponse.cpp
--- a/dom/fetch/InternalResponse.cpp	Tue Jan 17 15:13:44 2023 +0000
+++ b/dom/fetch/InternalResponse.cpp	Tue Jan 17 15:13:45 2023 +0000
@@ -186,6 +186,30 @@
   return result;
 }
 
+ParentToChildInternalResponse InternalResponse::ToParentToChildInternalResponse(
+    NotNull<mozilla::ipc::PBackgroundParent*> aBackgroundParent) {
+  ParentToChildInternalResponse result(GetMetadata(), Nothing(),
+                                       UNKNOWN_BODY_SIZE, Nothing());
+
+  nsCOMPtr<nsIInputStream> body;
+  int64_t bodySize;
+  GetUnfilteredBody(getter_AddRefs(body), &bodySize);
+
+  if (body) {
+    result.body() = Some(
+        ToParentToChildStream(WrapNotNull(body), bodySize, aBackgroundParent));
+    result.bodySize() = bodySize;
+  }
+
+  nsCOMPtr<nsIInputStream> alternativeBody = TakeAlternativeBody();
+  if (alternativeBody) {
+    result.alternativeBody() = Some(ToParentToChildStream(
+        WrapNotNull(alternativeBody), UNKNOWN_BODY_SIZE, aBackgroundParent));
+  }
+
+  return result;
+}
+
 SafeRefPtr<InternalResponse> InternalResponse::Clone(CloneType aCloneType) {
   SafeRefPtr<InternalResponse> clone = CreateIncompleteCopy();
   clone->mCloned = (mCloned = true);